lib/module.js

```diff
@@ -120,12 +121,7 @@ function tryExtensions(p, exts) {
   return false;
 }
 
-
-const noopDeprecateRequireDot = util.deprecate(function() {},
-    'warning: require(\'.\') resolved outside the package directory. ' +
-    'This functionality is deprecated and will be removed soon.');
-
-
+var warned = false;
```

Comment:
Not a fan of this extra variable. Maybe `printDeprecationMessage` can be refactored to track which strings were printed, so we can remove this variable and argument altogether? cc: @vkurchatkin